src/compiler/emitter.ts

```diff
@@ -4564,15 +4564,15 @@ namespace ts {
                     // JsxText will be written with its leading whitespace, so don't add more manually.
                     return 0;
                 }
-                else if (preserveSourceNewlines && siblingNodePositionsAreComparable(previousNode, nextNode)) {
-                    return getEffectiveLines(
-                        includeComments => getLinesBetweenRangeEndAndRangeStart(
-                            previousNode,
-                            nextNode,
-                            currentSourceFile!,
-                            includeComments));
-                }
-                else if (!preserveSourceNewlines && !nodeIsSynthesized(previousNode) && !nodeIsSynthesized(nextNode)) {
+                else if (siblingNodePositionsAreComparable(previousNode, nextNode)) {
+                    if (preserveSourceNewlines) {

```

Comment:
To avoid incurring a performance penalty during normal program emit (i.e., via `tsc`), `siblingNodePositionsAreComparable` must not be called unless `preserveSourceNewlines` is true (which only happens in the language service for determining changes for codefixes and refactors).